Fix #1314 event in onAddClick is no longer mandatory - with test

Parro opened the pull request:
### Reasons for making this change

Calling `onAddClick` from a custom `ArrayFieldTemplate` gives error if `event` is not passed as parameter, but there are several cases in which `event` is not set. This PR makes `event` optional fixing #1314. Code update taken from #1365 and #1324.

### Checklist

* [ ] **I'm updating documentation**
  - [ ] I've [checked the rendering](https://react-jsonschema-form.readthedocs.io/en/latest/#contributing) of the Markdown text I've added
* [x] **I'm adding or updating code**
  - [x] I've added and/or updated tests
  - [x] I've updated [docs](https://react-jsonschema-form.readthedocs.io/) if needed
* [ ] **I'm adding a new feature**
  - [ ] I've updated the playground with an example use of the feature

epicfaace commented on docs/advanced-customization.md:
```diff
@@ -120,7 +120,7 @@ The following props are passed to each `ArrayFieldTemplate`:
 - `idSchema`: Object
 - `items`: An array of objects representing the items in the array. Each of the items represent a child with properties described below.
-- `onAddClick: (event) => void`: A function that adds a new item to the array.
+- `onAddClick: ([event]) => void`: A function that adds a new item to the array. `event` is optional.
```
```suggestion
- `onAddClick: (event?) => void`: A function that adds a new item to the array. `event` is optional.
```

epicfaace commented on packages/core/test/ArrayField_test.js:
```diff
@@ -249,6 +273,18 @@ describe("ArrayField", () => {
+
+      Simulate.click(node.querySelector(".array-item-add button"));
+
+      expect(node.querySelector(".array-item").hasAttribute(ArrayKeyDataAttr))
```
Can you instead make the test check just that `.array-item` exists? I don't think we need to check if the key is present, so we should safely be able to remove that check and the data-rjsf-itemkey attribute (which was only a hack needed to check the keys of array items for specific tests regarding array reordering)

Parro replied:
I fixed the test but I've never used chai... Did i used the right syntax?

Parro commented:
> Thanks! Can you also update the types here: https://github.com/rjsf-team/react-jsonschema-form/blob/master/packages/core/index.d.ts#L184

Updated, I also removed the second event params, there is no currying for this function 😉

epicfaace commented on packages/core/test/ArrayField_test.js:
```diff
@@ -62,6 +62,30 @@ const ExposedArrayKeyTemplate = function(props) {
+          <div
+            key={element.key}
+            className="array-item"
+            data-rjsf-itemkey={element.key}>
```
Can you remove this attribute? then we should be good

Parro replied:
Fixed
